plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/PipelineStore.java

```diff
@@ -60,73 +63,82 @@
+    public PipelineStore(Settings settings, Provider<Client> clientProvider, Environment environment, Map<String, ProcessorFactoryProvider> processorFactoryProviders) {
         super(settings);
-        this.threadPool = threadPool;
-        this.clusterService = clusterService;
         this.clientProvider = clientProvider;
         this.scrollTimeout = settings.getAsTime("ingest.pipeline.store.scroll.timeout", TimeValue.timeValueSeconds(30));
-        this.pipelineUpdateInterval = settings.getAsTime("ingest.pipeline.store.update.interval", TimeValue.timeValueSeconds(1));
         Map<String, Processor.Factory> processorFactories = new HashMap<>();
         for (Map.Entry<String, ProcessorFactoryProvider> entry : processorFactoryProviders.entrySet()) {
             Processor.Factory processorFactory = entry.getValue().get(environment);
             processorFactories.put(entry.getKey(), processorFactory);
         }
         this.processorFactoryRegistry = Collections.unmodifiableMap(processorFactories);
-        clusterService.add(new PipelineStoreListener());
     }
 
     @Override
-    protected void doStart() {
-    }
-
-    @Override
-    protected void doStop() {
-    }
-
-    @Override
-    protected void doClose() {
-        // TODO: When org.elasticsearch.node.Node can close Closable instances we should remove this code
+    public void close() throws IOException {
+        // TODO: When org.elasticsearch.node.Node can close Closable instances we should try to remove this code,
+        // since any wired closable should be able to close itself
         List<Closeable> closeables = new ArrayList<>();
         for (Processor.Factory factory : processorFactoryRegistry.values()) {
             if (factory instanceof Closeable) {
                 closeables.add((Closeable) factory);
             }
         }
-        try {
-            IOUtils.close(closeables);
-        } catch (IOException e) {
-            throw new RuntimeException(e);
-        }
+        IOUtils.close(closeables);
     }
 
     /**
      * Deletes the pipeline specified by id in the request.
      */
     public void delete(DeletePipelineRequest request, ActionListener<DeleteResponse> listener) {
+        ensureReady();
+
         DeleteRequest deleteRequest = new DeleteRequest(request);
         deleteRequest.index(PipelineStore.INDEX);
         deleteRequest.type(PipelineStore.TYPE);
         deleteRequest.id(request.id());
         deleteRequest.refresh(true);
-        client().delete(deleteRequest, listener);
+        client().delete(deleteRequest, new ActionListener<DeleteResponse>() {
+            @Override
+            public void onResponse(DeleteResponse deleteResponse) {
+                client().execute(RefreshAction.INSTANCE, new RefreshRequest(), new ActionListener<RefreshResponse>() {
```

Comment:
ok...but client depends on the transport service anyway no? I think I don't get it